contentcuration/contentcuration/viewsets/channel.py

```diff
@@ -505,26 +507,37 @@ def get_queryset(self):
 
     def annotate_queryset(self, queryset):
         queryset = super(AdminChannelViewSet, self).annotate_queryset(queryset)
-
-        editor_query = (
-            User.objects.filter(editable_channels__id=OuterRef("id"))
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        viewers_query = (
-            User.objects.filter(view_only_channels__id=OuterRef("id"))
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        file_query = (
-            File.objects.filter(contentnode__tree_id=OuterRef("main_tree__tree_id"))
-            .values("checksum", "file_size")
-            .distinct()
-        )
-        queryset = queryset.annotate(
-            editors_count=SQCount(editor_query, field="id"),
-            viewers_count=SQCount(viewers_query, field="id"),
-            size=SQSum(file_query, field="file_size"),
-        )
-
         return queryset
+
+    def consolidate(self, items, queryset):
+        if items:
+            for item_channel in items:
+                metadata = self.get_or_cache_channel_metadata(
+                    item_channel["id"], item_channel["main_tree__tree_id"]
+                )
+                if metadata == DEFERRED_FLAG:
+                    item_channel["size"] = item_channel["editors_count"] = item_channel[
+                        "editors_count"
+                    ] = DEFERRED_FLAG
+                else:
+                    item_channel["size"] = metadata["SIZE"]
+                    item_channel["editors_count"] = metadata["EDITORS"]
+                    item_channel["viewers_count"] = metadata["VIEWERS"]
+        return items
+
+    def get_or_cache_channel_metadata(self, channel, tree_id):
+        key = "channel_metadata_{}".format(channel)
```

Comment:
Should this key live somewhere else, as I assume it gets referenced in the metadata generation task too?